LucidTests/Utils/AsyncTaskQueueTests.swift

```diff
@@ -0,0 +1,316 @@
+//
+//  AsyncTaskQueueTests.swift
+//  LucidTests
+//
+//  Created by Stephane Magne on 2023-05-12.
+//  Copyright © 2023 Scribd. All rights reserved.
+//
+
+import XCTest
+@testable import Lucid
+
+final class AsyncTaskQueueTests: XCTestCase {
+
+    // MARK: Adding to empty queue
+
+    func test_that_it_immediately_runs_a_concurrent_operation_added_to_an_empty_queue() {
```

Comment:
If you need to test async functions, you can make the test func `async` instead of using multiple tasks. Just a tip there

Reply:
So, I've been playing around with that and correct me if I'm wrong here, but I see it as:

- `async` test, good for testing a call to a single `async` function
- standard test, good to simulate calling from multiple threads (setting up multiple Task blocks).

I started by making these `async` functions, but I was having trouble not getting them to deadlock because my tests were all waiting on the same initial `async` thread.

I tried switching them over to `TaskGroups` but I was still having some issues there. 

So, that's how I ended up with my bullet points above. I'm not sure if that thinking is 100% correct or not. I just wasn't able to express to myself the advantage of using an `async` function when I wanted to simulate calling from multiple threads.

Reply:
If you need to perform tasks in parallel you can always use the Task pattern you are using. If you make the function async, you would be able to remove one of those Tasks and keep the main flow of the test within it's own context